juju/juju
# CancelGeneration apiserver piece

## Description of change

- Add CancelGeneration to apiserver for modelgeneration.
- Unit tests for api/modelgeneration added for  CancelGeneration.
- Updated cmd and api for CancelGeneration to take model tags.

## QA steps

1. export JUJU_DEV_FEATURE_FLAGS=generations
2. juju bootstrap localhost
2052.  juju deploy percona-cluster mysql -n 2
2056.  juju add-generation
2057.  juju cancel-generation  <-- should succeed
2056.  juju add-generation
2063.  juju advance mysql/0
2057.  juju cancel-generation  <-- should fail and list mysql/1 as the reason.
3. juju switch-generation current
2057.  juju cancel-generation  <-- should fail
4. run `juju-db.bash`
2. db.generations.find(), should report 2 docs with: active = false, completed with a non zero NumberLong.

## Documentation changes

In the future
